library/src/test/java/SpecMathTest.java

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import org.junit.jupiter.api.Test;
+
+class SpecMathTest {
+  @Test
+  void testUnionWithoutConflicts()
+      throws IOException, UnionConflictException, UnexpectedTypeException {
+    String spec1String = Files.readString(Path.of("src/test/resources/noConflict1.yaml"));
+    String spec2String = Files.readString(Path.of("src/test/resources/noConflict2.yaml"));
+    String actual = SpecMath.union(spec1String, spec2String);
+    String expectedString = Files.readString(Path.of("src/test/resources/noConflictMerged.yaml"));
+
+    assertThat(actual).isEqualTo(expectedString);
+  }
+
+  @Test
+  void testUnionWithConflictResolutionsAndDefaults()
+      throws IOException, UnionConflictException, UnexpectedTypeException {
+    String spec1String = Files.readString(Path.of("src/test/resources/conflict1.yaml"));
```

Comment:
It doesn't look like any of the calls to other classes are mocked.

Reply:
As we discussed, the SpecMathTest will be added to a directory of integration tests. This will be done in a PR after these are merged to change the directory structure of the whole library.